RDI: adding write-behind configuration section (#2530)
RedisLabs/redislabs-docs: content/rdi/write-behind/configuration-guide.md

---
Title: Write-behind configuration guide
linkTitle: Write-behind-configuration
description: Configuring write-behind to your database
weight: 40
alwaysopen: false
categories: ["redis-di"]
aliases: 
draft: 
hidden: false
---

This guide provides details about configuring write-behind target connections.

## Overview

> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database. The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.

Reviewer: @redima I need a little clarification about this sentence: "Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database." It's unclear to me what "to when they are captured" means.

> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database. The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.

Author: sure, perhaps need a comma to read it like below:
> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to, when they are captured in RDI...

> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database. The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.

Author: or maybe:
> ... after they are captured in RDI

> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database. The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.

Reviewer 2: ```suggestion
Write-behind target connections are the connections established between the RDI instance and the target database in a [write-behind scenario]({{</* relref  "/rdi/quickstart/write-behind-guide.md" */>}})`. They replicate the changes captured in an RDI-enabled Redis Enterprise database to the target database. 
The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.
```

@redima

> Write-behind target connections are used in RDI write-behind jobs (commonly referred as recipes) to replicate changes to when they are captured in a RDI-enabled Redis Enterprise database. The connections must be configured in the `config.yaml` before deploying any jobs and must follow one of the formats shown below. Multiple connections can be specified in the `connections` section at the same time.

Reviewer 2: @redima  It didn't feel like the connection configuration section is the best one to introduce the recipes. I think they might be a better fit for the quickstart section, and we link to it from here. What do you think?

```yaml
connections:
    mssql2019-sqlauth:
        type: sqlserver
        host: ip-10-0-0-5.internal
        port: 1433
        database: rdi_wb_database
        user: rdi_user
        password: secret
```

### Windows Authentication

To use Windows Authentication mode, you need to create a Windows or AD account that has the necessary permissions to access the target database and is able to log into SQL Server. The Linux machine hosting RDI can be configured to support NTLM or Kerberos authentication protocols. Please consult with your security team as the configuration settings will vary.

**For NTLM:**
```yaml
connections:
    mssql2019-ntlm:
        type: sqlserver
        host: ip-10-0-0-5.internal
        port: 1433
        database: rdi_wb_database
        user: MYDOMAIN\rdi_service_account  # company-domain\service-account
        password: secret                    # NTLM requires to provide a password
```

> Note: User must be specified with the domain name for Windows Authentication to work

**For Kerberos:**

> * Please contact Redis team

Reviewer: @redima Consider adding the contact information

Author: @yaronp68 please take a look - what do we want to put in here?

> After configuring the RDI connection and deploying the write-behind job, the operator can check whether RDI is using the expected authentication mechanism and protocol by running the following SQL query (note, this may require `sysadmin` role):

Reviewer: ```suggestion
After you configure the RDI connection and deploy the write-behind job, run the following SQL query to have the operator check if RDI is using the expected authentication mechanism and protocol. **Note:** This operation may require `sysadmin` role.
```

```sql
SELECT session_id, auth_scheme FROM sys.dm_exec_connections;
```

> The results will indicate which `auth_scheme` is used by each session and may take the following values: `SQL`, `NTLM` and `Kerberos`.

Reviewer: ```suggestion
The results indicate which `auth_scheme` is used by each session and may take values `SQL`, `NTLM`, and `Kerberos`.
```

### Azure Active Directory

> * Please contact Redis team

Reviewer: @redima consider adding contact info

Author: same as above. let's see what Yaron will reply with.

Reviewer 3: @redima please remove the kerberos & Azure AD section (or say not supported for now)
